libres/lib/analysis/update.cpp

```diff
@@ -446,96 +482,123 @@ void assert_size_equal(int ens_size, const bool_vector_type *ens_mask) {
+                                  local_ministep_type *ministep,
+                                  rng_type *shared_rng, FILE *log_stream) {
+    /*
+    Observations and measurements are collected in these temporary
+    structures. obs_data is a precursor for the 'd' vector, and
+    meas_data is a precursor for the 'S' matrix'.
+
+    The reason for going via these temporary structures is to support
+    deactivating observations which should not be used in the update
+    process.
+    */
+    double alpha = analysis_config_get_alpha(analysis_config);
+    double std_cutoff = analysis_config_get_std_cutoff(analysis_config);
+    double global_std_scaling =
+        analysis_config_get_global_std_scaling(analysis_config);
+
+    obs_data_type *obs_data = obs_data_alloc(global_std_scaling);
+    meas_data_type *meas_data = meas_data_alloc(ens_mask);
+
+    enkf_analysis_fprintf_obs_summary(
+        obs_data, meas_data, local_ministep_get_name(ministep), log_stream);
+
+    int_vector_type *ens_active_list = bool_vector_alloc_active_list(ens_mask);
+
+    local_obsdata_type *selected_observations =
+        local_ministep_get_obsdata(ministep);
+    enkf_obs_get_obs_and_measure_data(obs, source_fs, selected_observations,
+                                      ens_active_list, meas_data, obs_data);
+    int_vector_free(ens_active_list);
+
+    enkf_analysis_deactivate_outliers(obs_data, meas_data, std_cutoff, alpha,
+                                      true);
+
+    if (meas_data_get_active_obs_size(meas_data) == 0) {
+        obs_data_free(obs_data);
+        meas_data_free(meas_data);
+        return update_data_type();
+    }
+
+    int active_ens_size = meas_data_get_active_ens_size(meas_data);
+    auto S = meas_data_allocS(meas_data);
+    meas_data_free(meas_data);
+
+    matrix_type *E = obs_data_allocE(obs_data, shared_rng, active_ens_size);
+    int_vector_type *iens_active_index =
+        bool_vector_alloc_active_index_list(ens_mask, -1);
+    auto A = load_parameters(target_fs, ensemble_config, iens_active_index,
+                             active_ens_size, ministep);
+
+    int active_obs_size = obs_data_get_active_size(obs_data);
+    matrix_type *R = obs_data_allocR(obs_data);
+    matrix_type *D = obs_data_allocD(obs_data, E, S);
+    assert_matrix_size(E, "E", active_obs_size, active_ens_size);
+    assert_matrix_size(D, "D", active_obs_size, active_ens_size);
+    assert_matrix_size(S, "S", active_obs_size, active_ens_size);
+    assert_matrix_size(R, "R", active_obs_size, active_obs_size);
+    const bool_vector_type *obs_mask = obs_data_get_active_mask(obs_data);
+    obs_data_scale(obs_data, S, E, D, R, nullptr);
+
+    auto row_scaling_parameters = load_row_scaling_parameters(
+        target_fs, ensemble_config, iens_active_index, active_ens_size,
+        ministep);
+
+    local_ministep_add_obs_data(ministep,
+                                obs_data); // This is not correct conceptually
```

Comment:
What is not correct conceptually?

Reply:
ministep should only hold the configuration objects, like selected observations and parameters for the update step, not the actual data.

Reply:
That's a better comment I think.

Reply:
To me it looks like this is (partly ?) addressed in #2923